rust/windows-client/src-tauri/src/client.rs

```diff
@@ -37,30 +45,51 @@ pub(crate) struct GuiParams {
 /// `C:/Users/$USER/AppData/Local/dev.firezone.client`
 pub(crate) struct AppLocalDataDir(std::path::PathBuf);
 
-pub(crate) fn run() -> Result<()> {
-    // Special case for app link URIs
-    if let Some(arg) = std::env::args().nth(1) {
-        let url = url::Url::parse(&arg)?;
-        if url.scheme() == DEEP_LINK_SCHEME {
-            return gui::run(GuiParams {
-                inject_faults: false,
-            });
-        }
-    }
+// Hides Powershell's console on Windows
+// <https://stackoverflow.com/questions/59692146/is-it-possible-to-use-the-standard-library-to-spawn-a-process-without-showing-th#60958956>
+const CREATE_NO_WINDOW: u32 = 0x08000000;
 
+pub(crate) fn run() -> Result<()> {
     let cli = cli::Cli::parse();
 
     match cli.command {
-        None => gui::run(GuiParams {
-            inject_faults: cli.inject_faults,
-        }),
+        None => {
+            if elevation::check()? {
+                // We're already elevated, just run the GUI
+                gui::run(GuiParams {
+                    flag_elevated: false,
+                    inject_faults: cli.inject_faults,
+                })
+            } else {
+                // We're not elevated, ask Powershell to re-launch us, then exit
+                let current_exe = tauri_utils::platform::current_exe()?;
+                Command::new("powershell")
+                    .creation_flags(CREATE_NO_WINDOW)
+                    .arg("-Command")
+                    .arg("Start-Process")
+                    .arg("-FilePath")
+                    .arg(current_exe)
+                    .arg("-Verb")
+                    .arg("RunAs")
+                    .arg("-ArgumentList")
+                    .arg("elevated")
+                    .spawn()?;
```

Comment:
There's a bug here - If the exe path includes a space, like "Program Files", it gets split wrong when sending it to Powershell

Reply:
This one's probably gonna need more work cause of this

Reply:
Does it work if you just wrap in `"`? like `format!(\"{current_exe}\")`

Reply:
It'll probably work as long as the path doesn't somehow have quotes in it. It just didn't feel totally perfect

Reply:
If the user tries to stick Firezone in a path with quotes in it we shall unleash this pain unto them.